content/documentation/pages/2-concepts/1-architecture.md

```diff
@@ -1,33 +1,330 @@
+
+<!--NOTE-->
+
+By default, the REST endpoints (administration, management, and health) as well as the Dashboard UI do not require authenticated access.
+
+<!--END_NOTE-->
+
+## Application Types
+
+Applications come in two flavors:
+
+- Long-lived applications. There are two types of long-lived applications.
+
+  - Message-driven applications where an an unbounded amount of data is consumed or produced through a single input and/or output.
+  - The second is an message-driven application that can have multiple inputs and/or outputs. It could also be a application that does not use messaging middleware at all.
+
+- Short-lived applications that process a finite set of data and then terminate.  
+  There are two variations of short-lived applications.
+
+  - The first is a Task which executes your code and records the status of execution in the Data Flow database.
+    It can optionally use the Spring Cloud Task framework nor is it required for it to be a Java application.
+    However, the application does need to record it's execution status Data Flow's database.
+  - The second is an extension of the first which includes the Spring Batch framework as the foundation of performing batch processing.
+
+It is common to write long-lived applications based on the Spring Cloud Stream framework and short-lived applications based on the Spring Cloud Task or Spring Batch framework.  
+There are many guides in the documentation that show you how to use these frameworks in developing data pipleliens.
+However, it is also possible to write long-lived and short-lived applications that do not use Spring.
+They can also be written in other programming languages.
+
+Depending on the runtime, applications can be packaged in two ways:
+
+- A Spring Boot uber-jar that is can be accessed from a Maven repository, file location, or over http.
+- Docker image hosted in a Docker registry.
+
+### Long-lived applications
+
+Long-lived applications are expected to be continuously executing.
+If the application stops, the platform is responsible for restarting it.
+
+The Spring Cloud Stream framework provides a programming model to simplify the writing of message-driven microservice applications that are connected to a common messaging system.
+You write the core business logic that is agnostic to the specific middleware that will be used.
+The middleware to use is determined by adding a Spring Cloud Stream Binder library as a dependency to the application.
+There are binding libraries for the following messaging middleware products: [RabbitMQ](https://www.rabbitmq.com), [Kafka](https://kafka.apache.org), [Kafka Streams](https://kafka.apache.org/documentation/streams/), [Amazon Kinesis](https://aws.amazon.com/kinesis/), [Google Pub/Sub](https://cloud.google.com/pubsub/docs/), [Solace PubSub+](https://solace.com/software/), [Azure Event Hubs](https://azure.microsoft.com/en-us/services/event-hubs/).
```

Comment:
Maybe also, we can end with ".. and others" - we have rocketmq and we other variants of azure jms things